metadata_service/proxy/atlas_proxy.py

```diff
@@ -476,6 +479,95 @@ def get_table(self, *, table_uri: str) -> Table:
+            # is establishing which format out of the supported ones it is and then we validate every partition
+            # against it.
+            for df in app.config['WATERMARK_DATE_FORMATS']:
+                _, result = AtlasProxy._validate_date(partition_name, df)
+
+                if result:
+                    LOGGER.debug('Established date format', extra=dict(date_format=result))
+                    return result
+
+        return result
+
+    @staticmethod
+    def _render_partition_key_name(entity: EntityUniqueAttribute) -> Optional[str]:
+        _partition_keys = []
+
+        for partition_key in entity.get('attributes', dict()).get('partitionKeys', []):
+            partition_key_column_name = partition_key.get('displayName')
+
+            if partition_key_column_name:
+                _partition_keys.append(partition_key_column_name)
+
+        partition_key = ' '.join(_partition_keys).strip()
+
+        return partition_key
+
+    def _get_table_watermarks(self, entity: EntityUniqueAttribute) -> List[Watermark]:
+        partition_value_format = '%Y-%m-%d %H:%M:%S'
+
+        _partitions = entity.get('relationshipAttributes', dict()).get('partitions', list())
+        partition_key = AtlasProxy._render_partition_key_name(entity)
+
+        guids = [_partition.get('guid') for _partition in _partitions
+                 if _partition.get('entityStatus') == Status.ACTIVE
+                 and _partition.get('relationshipStatus') == Status.ACTIVE]
+
+        if not guids:
+            return []
+
+        full_partitions = extract_entities(self._driver.entity_bulk(guid=list(guids), ignoreRelationships=True))
+        watermark_date_format = AtlasProxy._select_watermark_format([p.attributes.get('name') for p in full_partitions])
+
+        _low_watermark_date, _high_watermark_date = None, None
+        low_watermark, high_watermark = None, None
+
+        for partition in full_partitions:
+            partition_name = partition.attributes.get('name')
+
+            if watermark_date_format and partition_name:
+                partition_date, _ = AtlasProxy._validate_date(partition_name, watermark_date_format)
+
+                if partition_date:
+                    _partition_create_time = self._parse_date(partition.createTime) or 0.0
+
+                    partition_create_time = datetime.datetime.fromtimestamp(
+                        _partition_create_time).strftime(partition_value_format)
+
+                    common_values = {'partition_value': datetime.datetime.strftime(partition_date,
+                                                                                   partition_value_format),
+                                     'create_time': partition_create_time,
+                                     'partition_key': partition_key}
+
+                    if (_low_watermark_date is None) or (partition_date < _low_watermark_date):
```

Comment:
Just use min() or max() to find the earliest or latest dates from a list. 
Append all the partition_dates in a list and apply min() and max() functions to get the `low_watermark` and `high_watermark`